### PR TITLE
Milestone #4 Implement add/remove Functionality

### PR DESCRIPTION
**Pull request by Anjacodes**
In this milestone, I 

- Implemented a function for adding a new task.
- Implemented a function for editing task descriptions.
- Implemented a function for deleting a task.
- Implemented a function for the "Clear all completed" button.

### CODE REVIEW

File: src/status.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+// import { clearCont } from './index.js';
```
Comment (Mihndim2020) [inline src/status.js]: - [ ]  Please kindly remove commented code from you code base as this makes you work dirty and less readable.

File: webpack.config.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 const path = require('path');
 const HmtlWebpackPlugin = require('html-webpack-plugin');
+// const { check } = require('lockfile');
```
Comment (Mihndim2020) [inline webpack.config.js]: - [ ]  Please kindly remove commented code from you code base as this makes you work dirty and less readable.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @Anjacodes, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Good job with Gitflow.

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

Please also consider:
- [ ] Your project is not running from my end, please kindly include in your `readme file` steps to clone and run you project locally. If possible add a `live link`. I am getting an unsupported platform error. If possible get another classmate/friend to clone and run it from their machine before you resubmit for re-review. See image below. 

unsupported

- [ ] The link to the screenshot is your readme file is broken, please kindly fix that.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/status.js (hunk 1 above)
Comment (Anjacodes) [reply src/status.js]: Thank you @Mihndim2020 for your suggestions! I removed these comments from my code.

File: webpack.config.js (hunk 2 above)
Comment (Anjacodes) [reply webpack.config.js]: Thank you @Mihndim2020 for your suggestions! I removed these comments from my code.

File: src/status.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        arr.forEach((task, i) => {
+          task.index = i;
+        });
+        localStorage.setItem('tasks', JSON.stringify(arr));
+        clearCont();
+        generateTasks(arr);
+      }
+    });
+  });
+};
```
Comment (omar-labana) [inline src/status.js:125]: - [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique (currently they maintainer their generated index).

File: src/status.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
+import { checkBox, clearCont } from './check.js';
+
+const addTask = document.querySelector('.subheading');
```
Comment (omar-labana) [inline src/status.js:3]: - [ ] Kindly make sure to clear the add task input field after successfully adding a task so users can add new tasks without clearing the field manually each time.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @Anjacodes, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
- [ ] The `dependencies` found in your `package.json` file is longest I have seen in my life so far. I think you accidentally installed or configured an unwanted `workframe` since all these dependencies has nothing related to `webpack`. Please remove all the packages found in the `dependencies` object found in the `package,json` file and add `webpack` to the `devDependencies` (currently missing). _That's why the previous CR couldn't start the project_

- [ ] _Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_N/A_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/status.js (hunk 3 above)
Comment (Anjacodes) [reply src/status.js:125]: Hi @omar-labana, thank you for your feedback 🙏 ! 
I'm not entirely sure as to what I should do as the index is actually being updated in lines 106/107 and 114/115. Let me share some screenshots below. Is it ok like this or do I need to change anything about it?
Thank you for confirming! 

Screen Shot 2021-12-03 at 7 41 09
Screen Shot 2021-12-03 at 7 41 18
Screen Shot 2021-12-03 at 7 41 29
?

File: src/status.js (hunk 4 above)
Comment (Anjacodes) [reply src/status.js:3]: Thank you, @omar-labana, for pointing this out! I had been meaning to implement that but totally forgot about it in the end. The input field now clears once the user types a task and hits enter. 👌

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @Anjacodes 

Amazing job implementing suggestions from the previous reviewer! 💯 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**